modules/client-common/src/main/java/org/apache/ignite/internal/jdbc/proto/event/JdbcQuerySingleResult.java

```diff
@@ -69,60 +73,61 @@ public JdbcQuerySingleResult() {
      */
     public JdbcQuerySingleResult(int status, String err) {
         super(status, err);
-
-        resultsAvailable = false;
     }
 
     /**
      * Constructor.
      *
-     * @param cursorId Cursor ID.
+     * @param cursorId Id of the cursor in case it was registered on server.
      * @param rowTuples Serialized SQL result rows.
      * @param columnTypes Ordered list of types of columns in serialized rows.
      * @param decimalScales Decimal scales in appearance order.
-     * @param last     Flag indicates the query has no unfetched results.
+     * @param hasMoreData Flag indicates the query has un-fetched results.
+     * @param hasNextResult Flag indicates that current result is part of multi-statement query, there is at least one more result.
      */
-    public JdbcQuerySingleResult(long cursorId, List<BinaryTupleReader> rowTuples, List<ColumnType> columnTypes, int[] decimalScales,
-            boolean last) {
-        super();
-
+    @SuppressWarnings("NullableProblems")
+    public JdbcQuerySingleResult(
+            @Nullable Long cursorId,
+            List<BinaryTupleReader> rowTuples,
+            List<ColumnType> columnTypes,
+            int[] decimalScales,
+            boolean hasMoreData,
+            boolean hasNextResult
+    ) {
         Objects.requireNonNull(rowTuples);
 
         this.cursorId = cursorId;
         this.rowTuples = rowTuples;
         this.columnTypes = columnTypes;
         this.decimalScales = decimalScales;
 
-        this.last = last;
-        this.isQuery = true;
+        this.hasMoreData = hasMoreData;
+        this.hasNextResult = hasNextResult;
 
-        hasResults = true;
-        resultsAvailable = true;
+        hasResultSet = true;
 
         assert decimalScales != null;
     }
 
     /**
      * Constructor.
      *
+     * @param cursorId Id of the cursor in case it was registered on server.
      * @param updateCnt Update count for DML queries.
+     * @param hasNextResult Flag indicates that current result is part of multi-statement query, there is at least one more result.
      */
-    public JdbcQuerySingleResult(long cursorId, long updateCnt) {
-        super();
-
+    public JdbcQuerySingleResult(@Nullable Long cursorId, long updateCnt, boolean hasNextResult) {
         this.updateCnt = updateCnt;
         this.cursorId = cursorId;
-
-        hasResults = false;
-        resultsAvailable = true;
+        this.hasNextResult = hasNextResult;
     }
 
     /**
      * Get the cursor id.
      *
      * @return Cursor ID.
      */
-    public Long cursorId() {
+    public @Nullable Long cursorId() {
```

Comment:
seems we need to check for null all places it touched?

Reply:
all invocations already null-safe